### DISCUSSION TITLE
UnityConvertersConfig asset not loaded when running Unity in batch mode?

### DISCUSSION THREAD

**Discussion by jchowdown** [applejag/Newtonsoft.Json-for-Unity#137] [Q&A] [answered]
Hi, we created and configured a UnityConvertersConfig instance to not use specific custom converters when deserializing JSONs. This appears to work wonderfully in windowed mode, but when we try to run the same code in a headless editor ("-batchmode"), our converters appear to be added to the global converter list and they get run for every field that we try to deserialize. Is there something we need to do to make the editor load the ScriptableObject while in headless mode?

EDIT: I forgot to point out that this behavior is inconsistent cross machines. It works fine on my laptop but is broken (as described above) on our build machine (running Jenkins)

**Comment by applejag** [comment]
Hello @jchowdown! Thanks for reporting this, I've not heard about any use cases for headless editor yet, so I've sadly not considered it yet.

During initialization, it tries to load the Unity resource asset by the name `"Newtonsoft.Json-for-Unity.Converters"` using [`UnityEngine.Resources.Load()`](https://docs.unity3d.com/ScriptReference/Resources.Load.html), and if it fails it will [create a new UnityConvertersConfig](https://github.com/jilleJr/Newtonsoft.Json-for-Unity.Converters/blob/1.1.1/UnityConverters/UnityConverterInitializer.cs#L114-L133) using [default values](https://github.com/jilleJr/Newtonsoft.Json-for-Unity.Converters/blob/1.1.1/Configuration/UnityConvertersConfig.cs#L15-L30).

The only conclusion I can make is that your Jenkins build may have the `Resources` folder missing, but that seams unplausible since it probably got the other files loaded/cloned properly.

Are you also using the `-executeMethod` argument?
And what version of Unity, Newtonsoft.Json-for-Unity (or other Newtonsoft.Json package), and Newtonsoft.Json-for-Unity.Converts?

Either case, I'll do some manual testing to try and reproduce this. I've opened an issue of this over at https://github.com/jilleJr/Newtonsoft.Json-for-Unity.Converters/issues/61

**Comment by applejag** [comment, marked as answer]
I was unable to reproduce this issue, sadly. But let's move this discussion over to the issue https://github.com/jilleJr/Newtonsoft.Json-for-Unity.Converters/issues/61 as this is a new issue that I don't have an answer for (yet), so a GitHub Issue is better suited than a GitHub Discussion.
